WolvenKit.App/Services/UpdateService.cs

```diff
@@ -0,0 +1,151 @@
+
+namespace WolvenKit.App.Services;
+
+public class UpdateService : IUpdateService
+{
+    private readonly ILoggerService _loggerService;
+    private readonly ISettingsManager _settingsManager;
+    private readonly HttpClient _httpClient;
+    private readonly GitHubClient _gitHubClient;
+    
+    public UpdateService(ILoggerService loggerService, ISettingsManager settingsManager)
+    {
+        _loggerService = loggerService;
+        _settingsManager = settingsManager;
+        _httpClient = new HttpClient();
+        _httpClient.DefaultRequestHeaders.Add("User-Agent", "WolvenKit");
+        _gitHubClient = new GitHubClient(new ProductHeaderValue("WolvenKit"));
+    }
+    
+    public async Task UpdateToNewestVersion()
+    {
+        if (!await IsUpdateAvailable())
+        {
+            return;
+        }
+        
+        var latestRelease = await GetLatestRelease();
+        if (latestRelease is null)
+        {
+            return;
+        }
+        
+        var downloadZipPath = Path.Join(ISettingsManager.GetTemp_DownloadsPath(), latestRelease.TagName + ".zip");
+        var portableAsset = latestRelease.Assets.First(a => a.Name == $"WolvenKit-{latestRelease.TagName}.zip");
+        
+        try
+        {
+            var responseAsset = await _httpClient.GetAsync(portableAsset.BrowserDownloadUrl);
+            responseAsset.EnsureSuccessStatusCode();
+            await File.WriteAllBytesAsync(downloadZipPath, await responseAsset.Content.ReadAsByteArrayAsync());
+        }
+        catch (HttpRequestException ex)
+        {
+            _loggerService.Error($"Failed to download update: {portableAsset.BrowserDownloadUrl}");
+            _loggerService.Error(ex);
+            return;
+        }
+        
+        var unzipPath = Path.Join(ISettingsManager.GetTemp_DownloadsPath(), "unzip");
+        Directory.CreateDirectory(unzipPath);
+        ZipFile.ExtractToDirectory(downloadZipPath, unzipPath);
+        
+        var exePath = Environment.ProcessPath ?? Path.Combine(AppContext.BaseDirectory, "WolvenKit.exe");
+        var scriptPath = Path.Combine(ISettingsManager.GetTemp_DownloadsPath(), "update.ps1");
+        var vbsScriptPath = Path.Combine(ISettingsManager.GetTemp_DownloadsPath(), "update.vbs");
+        File.WriteAllText(scriptPath, $@"
+$exePath = ""{exePath}""
+$unzipPath = ""{unzipPath}""
+$rootTempPath = ""{ISettingsManager.GetTemp_DownloadsPath()}""
+
+while (Get-Process -Name (Split-Path $exePath -LeafBase) -ErrorAction SilentlyContinue) {{
+    Start-Sleep -Seconds 1
+}}
+
+Copy-Item -Path ""$unzipPath\*"" -Destination $appBaseDir -Recurse -Force
+
+Start-Process -FilePath $exePath
+
+Remove-Item -Path $rootTempPath -Recurse -Force -ErrorAction SilentlyContinue
+
+");
+        
+        File.WriteAllText(vbsScriptPath, $@"Set objShell = CreateObject(""WScript.Shell"")
+objShell.Run ""powershell.exe -ExecutionPolicy Bypass -File """"{scriptPath}"""""", 0, False
+");
+        Process.Start(new ProcessStartInfo
```

Comment:
Have you tested this on systems where the Execution Policy for Powershell scripts was at a higher security level?

Reply:
Hmmm I see that now. While I know it seems advantageous for us to use it, this would seem like a pretty big security flaw if that's the case. Seems to be it though... Alright